lib/Deserializer/functions.php

```diff
@@ -137,7 +150,17 @@ function enum(Reader $reader, string $namespace = null) : array {
         $reader->next();
         return [];
     }
-    $reader->read();
+    if (!$reader->read()) {
+        $reader->next();
+
+        return [];
+    }
+
+    if (Reader::END_ELEMENT === $reader->nodeType) {
+        $reader->next();
+
+        return [];
+    }
```

Comment:
This feels a bit hacky. Makes me wonder if this is there is a cleaner way to do this. Would for instance just a single `$reader->next()` do the job (removing `$reader->read()`?)

Reply:
THX for your feedback @evert 

I can have a look at this .... 
Since I already tagged the versions are you okay in accepting this implementation as is?

Reply:
Yea np. If it works it works. It was a mild concern. The tests pass which is the most important =)